### PR TITLE
Add TokenTotalSupplyUpdater

### PR DESCRIPTION
**Pull request by Qwerty5Uiop**
## Motivation

Token's `total_supply` update happens on every transfer to or from burn address and since token transfers are indexed concurrently, there may be many updates of the same tokens at the same time.

## Changelog

Added `TokenTotalSupplyUpdater` which accumulates unique tokens to update and updates them together periodically

### CODE REVIEW

**Review by vbaranov — APPROVED** [review]

File: apps/indexer/lib/indexer/fetcher/token_total_supply_updater.ex [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    token = Repo.get_by(Token, contract_address_hash: address_hash)
+
+    if token && !token.skip_metadata do
+      token_params =
+        address_hash_string
+        |> MetadataRetriever.get_total_supply_of()
+
+      token_to_update =
+        token
+        |> Repo.preload([:contract_address])
```
Comment (sl1depengwyn) [inline apps/indexer/lib/indexer/fetcher/token_total_supply_updater.ex:58]: Do we really need this preload?

Comment (sl1depengwyn) [inline apps/indexer/lib/indexer/fetcher/token_total_supply_updater.ex:58]: Do we really need this preload?

File: apps/indexer/lib/indexer/fetcher/token_total_supply_updater.ex [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  Periodically updates tokens total_supply
+  """
+
+  use GenServer
+
+  alias Explorer.{Chain, Repo}
+  alias Explorer.Chain.Token
+  alias Explorer.Token.MetadataRetriever
+
+  @update_interval :timer.seconds(10)
```
Comment (nikitosing) [inline apps/indexer/lib/indexer/fetcher/token_total_supply_updater.ex:12]: what if use for update_interval avg block time and as fallback 10 seconds for example

File: apps/indexer/lib/indexer/fetcher/token_total_supply_updater.ex (hunk 1 above)
Comment (Qwerty5Uiop) [reply apps/indexer/lib/indexer/fetcher/token_total_supply_updater.ex:58]: seems like we don't

**Review by sl1depengwyn — APPROVED** [review]

**Review by nikitosing — APPROVED** [review]

**Pull request merged**